### PR TITLE
🎉 feat(clipboard): allow to open meeting in clipboard

### PR DESCRIPTION
**Pull request by jgoldhammer**
### Status
**READY**

### Description
- we have introduced new functionality to access the clipboard
1. a new menu entry "Open meeting from clipboard"
2. open the link via shortcut

### Steps to Test or Reproduce
- define a shortcut for open link from clipboard
- Copy a link into the clipboard
- Use the shortcut or menu "open meeting from clipboard" 

image

### CODE REVIEW

**Comment by jgoldhammer** [comment]
What do you think @leits ?

**Comment by leits** [comment]
@jgoldhammer 
I think about this feature a lot.
The important thing in MB is privacy.
Therefore, I have a caveat against implicit monitoring, so I like the obvious action.
So LGTM.

It's out of scope, but I'm thinking of moving some actions to the "Quick Actions" submenu. 
It would be convenient to add a title hiding and skipping the next meeting there. WDYT?

File: MeetingBar/AppDelegate.swift [hunk 1]
```diff
@@ -118,6 +118,31 @@ class AppDelegate: NSObject, NSApplicationDelegate, UNUserNotificationCenterDele
...
+
+        if !clipboardContent.isEmpty {
+            let meetingLink = detectLink(&clipboardContent)
+            if meetingLink != nil {
+                openMeetingURL(meetingLink?.service, meetingLink!.url)
+            } else {
+                let validUrl = NSURL(string: clipboardContent)
+                if validUrl != nil {
+                    URL(string: clipboardContent)?.openInDefaultBrowser()
+                }
```
Comment (leits) [inline MeetingBar/AppDelegate.swift:138]: Maybe send a message if we did nothing with the clipboard?

File: MeetingBar/AppDelegate.swift [hunk 2]
```diff
@@ -118,6 +118,31 @@ class AppDelegate: NSObject, NSApplicationDelegate, UNUserNotificationCenterDele
...
+    func openLinkFromClipboard() {
+        NSLog("Check macos clipboard for link")
+        var clipboardContent = getClipboardContent()
+        NSLog("Found \(clipboardContent) in clipboard")
+
+        if !clipboardContent.isEmpty {
+            let meetingLink = detectLink(&clipboardContent)
+            if meetingLink != nil {
+                openMeetingURL(meetingLink?.service, meetingLink!.url)
+            } else {
```
Comment (leits) [inline MeetingBar/AppDelegate.swift:134]:
```suggestion
            if let meetingLink = meetingLink {
                openMeetingURL(meetingLink.service, meetingLink.url)
            } else {
```

**Comment by leits** [comment]
@jgoldhammer, please, take a look at the comment.

**Comment by jgoldhammer** [comment]
> @jgoldhammer
> I think about this feature a lot.
> The important thing in MB is privacy.
> Therefore, I have a caveat against implicit monitoring, so I like the obvious action.
> So LGTM.
> 
> It's out of scope, but I'm thinking of moving some actions to the "Quick Actions" submenu.
> It would be convenient to add a title hiding and skipping the next meeting there. WDYT?

Thanks for your opinion. Let us move this action when we work on the other feature to hide the title and skip next meeting.

File: MeetingBar/AppDelegate.swift (hunk 1 above)
Comment (jgoldhammer) [reply MeetingBar/AppDelegate.swift:151]: I have added a message.

File: MeetingBar/AppDelegate.swift (hunk 2 above)
Comment (jgoldhammer) [reply MeetingBar/AppDelegate.swift]: Changed.

**Review by leits — APPROVED** [review]

**Pull request merged**